app/models/contact.rb

```diff
@@ -12,7 +12,14 @@ class Contact < ApplicationRecord
   has_paper_trail
 
   validates :first_name, presence: true
+  validates :surname, presence: true
   validates_date :date_of_birth, allow_nil: true, allow_blank: true
+  validates :email, format: { with: URI::MailTo::EMAIL_REGEXP }, if: :email?
+  validates :telephone, numericality: true, length: { minimum: 9, maximum: 11 }, if: :telephone?
```

Comment:
the drawback is that we don't allow spaces any more for telephones

Reply:
yes, that does introduce some complexities.  It's a little tricky validating length and also allowing spaces